sdk/src/main/java/io/opentelemetry/sdk/metrics/ViewRegistry.java

```diff
@@ -47,39 +64,71 @@ Batcher createBatcher(
       MeterSharedState meterSharedState,
       InstrumentDescriptor descriptor) {
 
-    Aggregation aggregation = getRegisteredAggregation(descriptor);
+    ViewSpecification specification = findBestMatch(descriptor);
 
-    // todo: don't just use the defaults!
-    switch (descriptor.getType()) {
-      case COUNTER:
-      case UP_DOWN_COUNTER:
-      case SUM_OBSERVER:
-      case UP_DOWN_SUM_OBSERVER:
-        return Batchers.getCumulativeAllLabels(
-            descriptor, meterProviderSharedState, meterSharedState, aggregation);
-      case VALUE_RECORDER:
-        // TODO: Revisit the batcher used here for value observers,
-        // currently this does not remove duplicate records in the same cycle.
-      case VALUE_OBSERVER:
-        return Batchers.getDeltaAllLabels(
-            descriptor, meterProviderSharedState, meterSharedState, aggregation);
+    Aggregation aggregation = specification.aggregation();
+
+    if (Temporality.CUMULATIVE == specification.temporality()) {
```

Comment:
Think we'd usually order `specification.temporality() == ` or use a switch

Reply:
I've seen it both ways...I don't think there's much of a standard, or even agreement on this. Back in the day, I'd use `.equals()` even for enums, since someone could change it from an enum and break code if you use `==`.  :)